### PR TITLE
Documentation : Add Promtail's `file_sd_configs` scrap_configs sample

### PR DESCRIPTION
**Pull request by askya**
**What this PR does / why we need it**:
Add `file_sd_configs` scrap_configs sample.
Which will help users quickly figure out how to configure Promtail scrap log with file discovery. Especially those with no Prometheus background.
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/clients/promtail/scraping.md [hunk 1]
```diff
@@ -64,6 +64,29 @@ There are different types of labels present in Promtail:
   uniqueness of the streams. It is set to the absolute path of the file the line
   was read from.
 
+### Exemple of File Discovery
+```
```
Comment (JStickler) [inline docs/sources/clients/promtail/scraping.md:68]: Could you add a sentence here that explains the example?  What are we doing here? How can the reader use this example?  Your PR description says this is for promtail configuration, but you don't actually say that in the documentation.

File: docs/sources/clients/promtail/scraping.md [hunk 2]
```diff
@@ -64,6 +64,29 @@ There are different types of labels present in Promtail:
...
+scrape_configs:
+...
+ - job_name: apt-dpkg
+    file_sd_configs:
+      - files:
+          - /etc/promtail/dpkg-apt.yaml
+```
+
+In /etc/promtail/dpkg-apt.yaml
```
Comment (JStickler) [inline docs/sources/clients/promtail/scraping.md:77]: Same, there needs to be a sentence or two explaining this example.

**Review by JStickler — COMMENTED** [review]
[docs team]  You need to provide some context along with the examples.

**Comment by cstyan** [comment]
hi @askya please let us know if you want to finish off this PR by addressing Julie's comment above :point_up:

**Comment by askya** [comment]
Please @JStickler I make an update to fit your remarks.

File: docs/sources/clients/promtail/scraping.md [hunk 3]
```diff
@@ -64,6 +64,37 @@ There are different types of labels present in Promtail:
   uniqueness of the streams. It is set to the absolute path of the file the line
   was read from.
 
+### Example of File Discovery  
+To scrape a set of log files defined manually on a machine, one can use `static_configs` or `file_sd_cnfigs`. Using static_configs, one need to reload Promtail to apply modifications. Using `file_sd_configs` that reload is not needed. Promtail reloads discovery files when they are updated.
```
Comment (cstyan) [inline docs/sources/clients/promtail/scraping.md:68]: There's a misspelling here  `file_sd_cnfigs` -> `file_sd_configs`. I think also `one need to` should be `one needs to` or `one must`.

File: docs/sources/clients/promtail/scraping.md [hunk 4]
```diff
@@ -64,6 +64,37 @@ There are different types of labels present in Promtail:
...
+scrape_configs:
+ - job_name: apt-dpkg
+    file_sd_configs:
+      - files:
+        - /etc/promtail/dpkg-apt.yaml
+    refresh_interval: 5m
+```  
+The targets to be scraped by Promtail are defined in `/etc/promtail/dpkg-apt.yaml`. In fact, Promtail read the target to scrape in the list of file provided under `files`.  
+
+Bellow is the content of `/etc/promtail/dpkg-apt.yaml`.
```
Comment (cstyan) [inline docs/sources/clients/promtail/scraping.md:82]: Below

**Comment by cstyan** [comment]
Sorry @askya one more merge conflict, then we can get the build to pass and I will merge :+1:

**Comment by askya** [comment]
Please I do know exactly how to resolve the conflict.
Can you please provide me some steps @cstyan ?

**Comment by cstyan** [comment]
@askya sorry, just looks like you need to sign the CLA, everything else looks good: https://grafana.com/docs/grafana/latest/developers/cla/

**Comment by askya** [comment]
Please @cstyan  I signed the agreement but it is not being reported so. What can I do to fix that ?
I went to https://cla-assistant.io/check/grafana/loki?pullRequest=10162 as suggested in the link you provide in the previous message.

**Comment by cstyan** [comment]
hey @askya someone on my team pointed out that sometimes this can fail if the email address you've used for your github account is different than the one you've used to sign off commits. It also looks like two different git users (I assume both are you, just different config) made commits on your branch. Rebasing to just one commit for under the email associated with your github account is probably the easiest fix.

**Pull request closed without merging**